experimental/QuadFormAndIsom/src/enumeration.jl

```diff
@@ -0,0 +1,952 @@
+##################################################################################
+#
+# This is an import to Oscar of the methods written following the paper [BH23] on
+# "Finite subgroups of automorphisms of K3 surfaces".
+#
+##################################################################################
+
+##################################################################################
+#
+# Admissible triples
+#
+##################################################################################
+
+# The tuples in output are pairs of positive integers!
+function _tuples_divisors(d::T) where T <: Hecke.IntegerUnion
+  div = divisors(d)
+  return Tuple{T, T}[(dd,abs(divexact(d,dd))) for dd in div]
+end
+
+# This is line 8 of Algorithm 1, they correspond to the possible
+# discriminant for the genera A and B to glue to fit in C. d is
+# the determinant of C, m the maximal p-valuation of the gcd of
+# d1 and dp.
+function _find_D(d::T, m::Int, p::Int) where T <: Hecke.IntegerUnion
+  @hassert :ZZLatWithIsom 1 is_prime(p)
+  @hassert :ZZLatWithIsom 1 d != 0
+
+  # If m == 0, there are no conditions on the gcd of d1 and dp
+  if m == 0
+    return _tuples_divisors(d)
+  end
+  
+  D = Tuple{T, T}[]
+  # We try all the values of g possible, from 1 to p^m
+  for j=0:m 
+    g = p^j
+    dj = _tuples_divisors(d*g^2)
+    for (d1,dp) in dj
+      if mod(d1,g) == mod(dp,g) == 0
+        push!(D,(d1,dp))
+      end
+    end
+  end
+  return D
+end
+
+# This is line 10 of Algorithm 1. We need the condition on the even-ness of
+# C since subgenera of an even genus are even too. r is the rank of
+# the subgenus, d its determinant, s and l the scale and level of C
+function _find_L(pG::Int, nG::Int, r::Int, d::Hecke.RationalUnion, s::ZZRingElem, l::ZZRingElem, p::Hecke.IntegerUnion, even = true; pos::Int = -1)
+  L = ZZGenus[]
+  if r == 0 && d == 1
+    return ZZGenus[genus(integer_lattice(gram = matrix(QQ, 0, 0, [])))]
+  end
+  if pos >= 0
+    neg = r-pos
+    gen = integer_genera((pos, neg), d, even=even)
+    filter!(G -> Hecke.divides(numerator(scale(G)), s)[1], gen)
```

Comment:
```suggestion
    filter!(G -> divides(numerator(scale(G)), s)[1], gen)
```
Is there a reason why you use `Hecke.divides` instead of `divides`? Same change at other places

Reply:
I have a custom `divides` function in one of the file which creates conflict

Reply:
I could not find any `divides` in this PR. Maybe you mean `_divides`?

Reply:
Ah yes I forgot I renamed it indeed..